src/braket/aws/aws_device.py

```diff
@@ -671,29 +668,29 @@ def get_devices(
         return devices
 
     def _update_pulse_properties(self) -> None:
-        if hasattr(self.properties, "pulse") and isinstance(
+        if not hasattr(self.properties, "pulse") or not isinstance(
```

Comment:
This seems a little more difficult to read than just leaving this as a nested if statement. Maybe I'm being overly picky here but I thought I would call it out

Reply:
interesting catch. I think this early exit reduces the nesting of the following code, making it a bit more readable imo.